src/PrestaShopBundle/Controller/Admin/Sell/Catalog/CategoryController.php

```diff
@@ -416,6 +417,41 @@ public function deleteCoverImageAction(Request $request, $categoryId)
         ]);
     }
 
+    /**
+     * Deletes category thumbnail image.
+     *
+     * @param Request $request
+     * @param int $categoryId
+     *
+     * @return RedirectResponse
+     */
+    #[AdminSecurity("is_granted('update', request.get('_legacy_controller'))", message: 'You do not have permission to edit this.', redirectRoute: 'admin_categories_edit', redirectQueryParamsToKeep: ['categoryId'])]
+    public function deleteThumbnailImageAction(Request $request, $categoryId)
+    {
+        if (!$this->isCsrfTokenValid('delete-thumbnail-image', $request->request->get('_csrf_token'))) {
+            return $this->redirectToRoute('admin_security_compromised', [
+                'uri' => $this->generateUrl('admin_categories_edit', [
+                    'categoryId' => $categoryId,
+                ], UrlGeneratorInterface::ABSOLUTE_URL),
+            ]);
+        }
```

Comment:
Why would you handle this separately, the security is supposed to be handled by the framework automatically everywhere Why do you use a custom CSRF token here?

Reply:
Because category cover deletion mechanism uses it also.

Reply:
But why not use the regular CSRF token in this case? I don't remember any other use case where we need to customize the token this way, so it seems overkill to me 🤔

Reply:
I can do it Jo, I just did not know, that's all. Still learning. :-) I copied it from the cover mechanism.

Can you please suggest the code changes I need to do to achieve this and simplify this as you say? :-)

Reply:
In theory you simply don't need to add any override of the CSRF token, it will be automatically included in the URL And the security system will be able to detect if it's an incorrect one and redirect if needed

Reply:
Done :-)